lib/upsert/merge_function/Java_OrgPostgresqlJdbc4_Jdbc4Connection.rb

```diff
@@ -10,7 +10,7 @@ class Java_OrgPostgresqlJdbc4_Jdbc4Connection < MergeFunction
       def execute_parameterized(query, args = [])
         query_args = []
         query = query.gsub(/\$(\d+)/) do |str|
-          query_args << args[$1.to_i - 1]
+          query_args << args[Regexp.last_match[1].to_i - 1]
```

Comment:
i don't object to `$1`, i just thought it was available to you as the (normal-looking) block var `str` ... i am probably wrong, in any case, either of these is fine